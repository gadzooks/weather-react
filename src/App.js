import React from 'react';
import { mockWeatherForecast, mockWeatherForecastNormalized } from './api/mockData';
import './App.css';
import SearchableTableHook from './weather/forecast_summary/SearchableTableHook';
import LocationDetails from './weather/location_details/LocationDetails';
import { format, parse } from 'fecha';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoaded: false,
      error: null,
      forecast: {},
    }
  }

  componentDidMount() {
    // const result = mockWeatherForecastNormalized();
    // this.setState({

    // })
    fetch("http://localhost:4000/forecasts/mock", {mode:'cors'})
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            forecast: result.data,
            error: false,
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      )
  }

  render() {
    const { error, isLoaded, forecast } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>;
    } else if (!isLoaded) {
      return <div>Loading...</div>;
    } else {
      //TODO move this to a function
      const isWeekend = [];
      const dates = forecast.dates.map((d) => {
        const parsedDate = parse(d, 'YYYY-MM-DD');
        const dayOfWeek = format(parsedDate, 'ddd').toUpperCase();
        isWeekend.push(dayOfWeek === 'SUN' || dayOfWeek === 'SAT')
        return parsedDate;
      })

      //TODO return components only if all data is available
      return (
        <>
          <SearchableTableHook inputs={forecast} isWeekend={isWeekend} dates={dates} />
          <LocationDetails locationsByName={forecast.forecasts.byId} isWeekend={isWeekend} dates={dates} locations={forecast.locations.allIds} />
        </>
      )
    }
  }
}

export default App;
