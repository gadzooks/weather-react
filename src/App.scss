div.loading {
  display: flex;
  justify-content: center;
  padding-top: 20px;
}

div.loading img {
  width: 100%;
  max-width: 1000px;
}

/* variables */

/* reset */

/* base styles */

/* fonts */

/* mobile styles */

/* small table styles */
@media screen and (min-width: 620px) {
  
}

/* large tablets and laptop styles */
@media screen and (min-width: 960px) {
  
}
/* desktop styles */
@media screen and (min-width: 1200px) {
  
}