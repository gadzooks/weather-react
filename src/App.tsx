import React from 'react';
import './App.css';
import WeatherPage from './components/weather/WeatherPage';
import {
  BrowserRouter,
  Routes,
  Route,
  Link} from "react-router-dom";

function Home() {
  return <h2>Home</h2>;
}

export default function App() {
  return (
    <BrowserRouter>
      <div>
        <nav>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/forecasts/:dataSource">Forecasts</Link>
            </li>
          </ul>
        </nav>

        {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
        <Routes>
          <Route path="/" element={<Home/>}/>
          <Route path="/forecasts/:dataSource" element={<WeatherPage isLoaded={false} error={null} forecast={null}/>}/>
        </Routes>
      </div>
    </BrowserRouter>
  );
}