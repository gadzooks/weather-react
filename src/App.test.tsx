import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('show loading when waiting for data', () => {
  render(<App />);
  const linkElement = screen.getByText('Forecast (mock)');
  expect(linkElement).toBeInTheDocument();
});

it('renders a snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
