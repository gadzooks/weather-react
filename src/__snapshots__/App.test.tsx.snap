// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/forecasts/:dataSource"
          onClick={[Function]}
        >
          Forecasts
        </a>
      </li>
    </ul>
  </nav>
  <h2>
    Home
  </h2>
</div>
`;
